include/sys/time_units.h

```diff
@@ -11,6 +11,19 @@
 extern "C" {
 #endif
 
+/** @brief System-wide macro to denote "forever" in milliseconds
+ *
+ *  Usage of this macro is limited to APIs that want to expose a timeout value
+ *  that can optionally be unlimited, or "forever".
+ *  This macro can not be fed into kernel functions or macros directly. Use
+ *  @SYS_MSEC instead.
+ */
+#define SYS_FOREVER_MS  (-1) 
+
+/** @brief System-wide macro to convert milliseconds to kernel timeouts
+ */
+#define SYS_MSEC(ms) (ms == SYS_FOREVER_MS ? K_FOREVER : K_MSEC(ms)) 
```

Comment:
I believe this is what @joerchan was asking for

Reply:
This works for me.  I might suggest adding a note on use cases, pointing out that "milliseconds with forever" was the legacy timeout API and that this is for the benefit of subsystems that want to use that space and not k_timeout_t.

Reply:
Yes it is. Thank you.